src/amo/components/AddAddonToCollection/index.js

```diff
@@ -125,14 +125,14 @@ export class AddAddonToCollectionBase extends React.Component<InternalProps> {
 
   addToCollection(collection: CollectionType) {
     const { addon, currentUserId, dispatch, errorHandler } = this.props;
-    if (!addon) {
-      throw new Error(
-        'Cannot add to collection because no add-on has been loaded',
-      );
-    }
-    if (!currentUserId) {
-      throw new Error('Cannot add to collection because you are not signed in');
-    }
+    invariant(
```

Comment:
These are conditions that are impossible to be true for a user interacting with the component. This `addToCollection` function will never be called if there is no `addon` or no `currentUserId`. We still need these checks because Flow will get upset without them, as it doesn't know that we will never call this without an `addon` and a `currentUserId` in `this.props`.

`invariant` is a way of ensuring that a condition is true, and Flow understands it, but we don't add tests for calls to `invariant` in our test suites, so this is the proper way of doing this now. I also removed the tests from the test suite that check these exception conditions because it is impossible to test.

If we want to make sure that we have test coverage for these statements, then we could move this function outside of the component, make it accept `addon` and `currentUserId` as arguments, and write unit tests for it, but I don't think that's necessary.